server/app/services/ProductionService.ts

```diff
@@ -0,0 +1,94 @@
+import TYPES from '../constants/types';
+import { IProduction } from './../models/ProductionModel';
+import { inject, injectable } from 'inversify';
+import { ProductionRepository } from './../repository/ProductionRepository';
+import { BadRequestError, NotFoundError } from '../errors';
+import { BikeService } from './BikeService';
+import { PartService } from './PartService';
+import PartEntity from './../entities/Part';
+import BikeEntity from './../entities/Bike';
+
+@injectable()
+export class ProductionService {
+  constructor(
+    @inject(TYPES.ProductionRepository) private productionRepository: ProductionRepository,
+    @inject(TYPES.BikeService) private bikeService: BikeService,
+    @inject(TYPES.PartService) private partService: PartService
+  ) {}
+
+  /**
+   * Return all productions json objects
+   * @returns all productions
+   */
+  public async getProductions(): Promise<IProduction[]> {
+    return this.productionRepository.getList();
+  }
+
+  /**
+   * Create a new production json obj
+   * @param body production json obj
+   * @returns production json obj
+   */
+  public async createProduction(body: IProduction): Promise<IProduction> {
+    if (body.type === 'Part') {
+      const partEntity = {
+        name: body.componentDetail.name,
```

Comment:
```suggestion
        ...body.componentDetail,
        stock: body.quantity
```
can u try a little something like this

Reply:
not sure if its applicable here though